Connection Name template does not work

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

I am creating my personalized theme and i wanted to add a connection segment but the `{{.Name}}` template does not work!

here is the segment:
```
{  
    "type": "connection",  
    "style": "powerline",  
    "background": "#8822ee",  
    "foreground": "#222222",  
    "powerline_symbol": "\ue0b0",  
    "template": "{{ if eq .Type \"wifi\"}} {{.Name}} \uf1eb {{ else if eq .Type \"ethernet\"}}\ueba9 {{ end }}"  
}
```  

### Theme

I am using my own theme

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

powershell

### Log output

```Shell
Version: 23.14.2

Shell: powershell (5.1.22621.4249)

Prompt:

󰖳 RM99 ~ 98     10:49  ➜ ( main) 

Segments:

ConsoleTitle(false)                       -   2 ms
Path(true)                                -  10 ms
Battery(true)                             -  16 ms
Connection(true)                          -  16 ms
Time(true)                                -   8 ms
Git(true)                                 -  38 ms

Run duration: 149.1267ms

Cache path: C:\Users\Abel\AppData\Local\oh-my-posh

Config path: C:\Users\Abel\My-OMP.omp.json

Logs:

[DEBUG] 10:49:59.075 debug.go:49 → debug mode enabled
[DEBUG] 10:49:59.155 debug.go:49 → plain mode enabled
[DEBUG] 10:49:59.158 terminal.go:Getenv:164 → C:\Users\Abel\AppData\Local
[TRACE] 10:49:59.159 terminal.go:Getenv(LOCALAPPDATA) - 3.7353ms
[TRACE] 10:49:59.160 terminal.go:CachePath() - 5.0338ms
[DEBUG] 10:49:59.160 terminal.go:70 → loading cache file: C:\Users\Abel\AppData\Local\oh-my-posh\omp.cache
[DEBUG] 10:49:59.161 terminal.go:70 → loading cache key: upgrade_check
[TRACE] 10:49:59.161 terminal.go(C:\Users\Abel\AppData\Local\oh-my-posh\omp.cache) - 1.7488ms
[DEBUG] 10:49:59.161 terminal.go:Getenv:164 → C:\Users\Abel\AppData\Local
[TRACE] 10:49:59.161 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 10:49:59.161 terminal.go:CachePath() - 0s
[DEBUG] 10:49:59.161 terminal.go:70 → loading cache file: C:\Users\Abel\AppData\Local\oh-my-posh\omp.cache.ca0a95f1-8bc6-48c5-9e5d-1bb6f95c6177
[DEBUG] 10:49:59.162 terminal.go:70 → loading cache key: prompt_count_cache
[TRACE] 10:49:59.162 terminal.go(C:\Users\Abel\AppData\Local\oh-my-posh\omp.cache.ca0a95f1-8bc6-48c5-9e5d-1bb6f95c6177) - 605.8µs
[TRACE] 10:49:59.162 terminal.go:setPromptCount() - 0s
[DEBUG] 10:49:59.162 terminal.go:Getenv:164 → C:\Users\Abel\My-OMP.omp.json
[TRACE] 10:49:59.162 terminal.go:Getenv(POSH_THEME) - 0s
[DEBUG] 10:49:59.162 terminal.go:ResolveConfigPath:96 → config set using POSH_THEME: C:\Users\Abel\My-OMP.omp.json
[TRACE] 10:49:59.162 terminal.go:ResolveConfigPath() - 0s
[TRACE] 10:49:59.162 terminal.go:Init() - 89.9139ms
[TRACE] 10:49:59.162 terminal.go:Flags() - 0s
[TRACE] 10:49:59.164 load.go:loadConfig() - 1.7072ms
[TRACE] 10:49:59.164 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.165 debug.go:57 → terminal program: Windows Terminal
[DEBUG] 10:49:59.165 debug.go:57 → terminal shell: shell
[DEBUG] 10:49:59.165 terminal.go:Getenv:164 → NO DATA
[TRACE] 10:49:59.165 terminal.go:Getenv(OMP_CACHE_DISABLED) - 0s
[TRACE] 10:49:59.166 terminal_windows.go:WindowsRegistryKeyValue(HKEY_CURRENT_USER\Software\Microsoft\Windows\DWM\ColorizationColor) - 0s
[TRACE] 10:49:59.166 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.166 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.166 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.167 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.167 terminal_windows.go:WindowsRegistryKeyValue:215 → ColorizationColor(DWORD): 0xC40078D4
[TRACE] 10:49:59.168 terminal.go:Shell() - 0s
[DEBUG] 10:49:59.168 terminal.go:Getenv:164 → 5.1.22621.4249
[TRACE] 10:49:59.168 terminal.go:Getenv(POSH_SHELL_VERSION) - 0s
[DEBUG] 10:49:59.169 debug.go:PrintDebug:23 → segment: Title
[DEBUG] 10:49:59.170 text.go:Render:80 → rendering template:
[TRACE] 10:49:59.170 terminal.go:Flags() - 0s
[TRACE] 10:49:59.170 terminal.go:Shell() - 0s
[DEBUG] 10:49:59.170 terminal.go:Getenv:164 → 15
[TRACE] 10:49:59.170 terminal.go:Getenv(POSH_CURSOR_LINE) - 129.6µs
[DEBUG] 10:49:59.170 terminal.go:Getenv:164 → 1
[TRACE] 10:49:59.170 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 0s
[TRACE] 10:49:59.170 terminal.go:Flags() - 0s
[TRACE] 10:49:59.170 terminal.go:Flags() - 0s
[TRACE] 10:49:59.170 terminal.go:Flags() - 0s
[TRACE] 10:49:59.170 terminal.go:Flags() - 0s
[TRACE] 10:49:59.170 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.170 terminal.go:PathSeparator() - 0s
[TRACE] 10:49:59.170 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.170 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.170 terminal.go:Flags() - 0s
[TRACE] 10:49:59.170 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.175 terminal.go:Pwd:189 → C:\Users\Abel
[TRACE] 10:49:59.175 terminal.go:Pwd() - 4.3537ms
[DEBUG] 10:49:59.175 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.175 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.175 segment.go:SetEnabled:115 → segment: Git
[TRACE] 10:49:59.175 terminal.go:Pwd() - 0s
[TRACE] 10:49:59.175 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.175 terminal.go:Pwd() - 0s
[DEBUG] 10:49:59.175 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.175 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.175 terminal_windows.go:58 → C:\Users\Abel
[DEBUG] 10:49:59.175 segment.go:SetEnabled:115 → segment: Connection
[TRACE] 10:49:59.175 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.175 segment.go:SetEnabled:115 → segment: Battery
[DEBUG] 10:49:59.175 terminal.go:Getenv:164 → C:\Users\Abel\AppData\Local
[TRACE] 10:49:59.175 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 10:49:59.177 terminal.go:Pwd() - 0s
[DEBUG] 10:49:59.177 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.177 segment.go:SetEnabled:115 → segment: Path
[TRACE] 10:49:59.177 terminal_windows.go:IsWsl() - 0s
[TRACE] 10:49:59.177 terminal.go:Pwd() - 0s
[DEBUG] 10:49:59.177 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.177 path.go:149 → display_cygpath: false
[DEBUG] 10:49:59.177 segment.go:SetEnabled:115 → segment: Time
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:PathSeparator() - 0s
[TRACE] 10:49:59.177 terminal.go:Pwd() - 0s
[TRACE] 10:49:59.177 terminal.go:Shell() - 0s
[TRACE] 10:49:59.177 terminal.go:Shell() - 0s
[TRACE] 10:49:59.177 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.177 connection.go:Enabled:26 → wifi|ethernet
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.177 path.go:setMappedLocations:563 → mapped_locations_enabled: true
[DEBUG] 10:49:59.177 path.go:setMappedLocations:564 → 
[DEBUG] 10:49:59.177 path.go:setMappedLocations:565 → 
[DEBUG] 10:49:59.177 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:PathSeparator() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.177 path.go:setMappedLocations:566 → ~
[DEBUG] 10:49:59.177 path.go:setMappedLocations:571 → mapped_locations: map[]
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:PathSeparator() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:PathSeparator() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.177 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.177 path.go:colorizePath:740 → cycle: []
[DEBUG] 10:49:59.177 path.go:getFolderSeparator:273 → NO DATA
[DEBUG] 10:49:59.177 path.go:getFolderSeparator:275 → \
[DEBUG] 10:49:59.177 path.go:colorizePath:743 → cycle_folder_separator: false
[DEBUG] 10:49:59.177 path.go:colorizePath:744 → %s
[DEBUG] 10:49:59.177 path.go:colorizePath:746 → %s
[DEBUG] 10:49:59.177 path.go:colorizePath:747 → %s
[DEBUG] 10:49:59.177 path.go:colorizePath:748 → %s
[TRACE] 10:49:59.177 terminal.go:Pwd() - 0s
[DEBUG] 10:49:59.177 time.go:Enabled:29 → 15:04
[TRACE] 10:49:59.178 terminal_windows.go:Root() - 509.5µs
[TRACE] 10:49:59.178 terminal.go:Shell() - 0s
[TRACE] 10:49:59.178 terminal.go:StatusCodes() - 0s
[TRACE] 10:49:59.178 terminal_windows.go:IsWsl() - 0s
[DEBUG] 10:49:59.178 terminal.go:TemplateCache:651 → environment: [ALLUSERSPROFILE=C:\ProgramData APPDATA=C:\Users\Abel\AppData\Roaming ChocolateyInstall=C:\ProgramData\chocolatey CommonProgramFiles=C:\Program Files\Common Files CommonProgramFiles(x86)=C:\Program Files (x86)\Common Files CommonProgramW6432=C:\Program Files\Common Files COMPUTERNAME=PC-DI-ABEL ComSpec=C:\WINDOWS\system32\cmd.exe CONDA_PROMPT_MODIFIER=False DriverData=C:\Windows\System32\Drivers\DriverData ESET_OPTIONS=



                                         HOMEDRIVE=C: HOMEPATH=\Users\Abel LOCALAPPDATA=C:\Users\Abel\AppData\Local LOGONSERVER=\\PC-DI-ABEL NUMBER_OF_PROCESSORS=8 OneDrive=C:\Users\Abel\OneDrive - Deutsche Schule Genf OneDriveCommercial=C:\Users\Abel\OneDrive - Deutsche Schule Genf OneDriveConsumer=C:\Users\Abel\OneDrive OS=Windows_NT Path=C:\Program Files\Common Files\Oracle\Java\javapath;C:\Program Files (x86)\Common Files\Oracle\Java\java8path;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Intel\WiFi\bin\;C:\Program Files\Common Files\Intel\WirelessCommon\;C:\ProgramData\chocolatey\bin;C:\Program Files\Microsoft SQL Server\150\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn\;C:\Program Files\dotnet\;C:\Program Files\nodejs\;C:\Users\Abel\AppData\Local\Microsoft\WindowsApps;C:\Users\Abel\AppData\Local\Programs\Microsoft VS Code\bin;C:\Users\Abel\AppData\Local\Microsoft\WinGet\Packages\Schniz.fnm_Microsoft.Winget.Source_8wekyb3d8bbwe;C:\Users\Abel\AppData\Local\Programs\oh-my-posh\bin;C:\Users\Abel\AppData\Roaming\npm;C:\Users\Abel\AppData\Local\GitHubDesktop\bin;C:\Users\Abel\AppData\Local\Programs\Git\cmd PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC;.CPL POSH_CURSOR_COLUMN=1 POSH_CURSOR_LINE=15 POSH_INSTALLER=winget POSH_SESSION_ID=ca0a95f1-8bc6-48c5-9e5d-1bb6f95c6177 POSH_SHELL_VERSION=5.1.22621.4249 POSH_THEME=C:\Users\Abel\My-OMP.omp.json POSH_THEMES_PATH=C:\Users\Abel\AppData\Local\Programs\oh-my-posh\themes POWERLINE_COMMAND=oh-my-posh PROCESSOR_ARCHITECTURE=AMD64 PROCESSOR_IDENTIFIER=Intel64 Family 6 Model 142 Stepping 12, GenuineIntel PROCESSOR_LEVEL=6 PROCESSOR_REVISION=8e0c ProgramData=C:\ProgramData ProgramFiles=C:\Program Files ProgramFiles(x86)=C:\Program Files (x86) ProgramW6432=C:\Program Files PSModulePath=C:\Users\Abel\OneDrive - Deutsche Schule Genf\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules PUBLIC=C:\Users\Public SESSIONNAME=Console SystemDrive=C: SystemRoot=C:\WINDOWS TEMP=C:\Users\Abel\AppData\Local\Temp TMP=C:\Users\Abel\AppData\Local\Temp USERDOMAIN=PC-DI-ABEL USERDOMAIN_ROAMINGPROFILE=PC-DI-ABEL USERNAME=Abel USERPROFILE=C:\Users\Abel VBOX_MSI_INSTALL_PATH=C:\Program Files\Oracle\VirtualBox\ windir=C:\WINDOWS WSLENV=WT_SESSION:WT_PROFILE_ID: WT_PROFILE_ID={61c54bbd-c2c6-5271-96e7-009a87ff44bf} WT_SESSION=76cae455-0879-4120-9cd1-134afb43be05]
[TRACE] 10:49:59.178 terminal.go:Pwd() - 0s
[DEBUG] 10:49:59.178 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.178 terminal_windows.go:IsWsl() - 0s
[TRACE] 10:49:59.178 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.178 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.178 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.178 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.178 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.178 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.178 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.178 terminal.go:User:330 → Abel
[TRACE] 10:49:59.178 terminal.go:User() - 0s
[DEBUG] 10:49:59.179 terminal.go:Host:347 → PC-di-Abel
[TRACE] 10:49:59.179 terminal.go:Host() - 639.8µs
[TRACE] 10:49:59.179 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.179 terminal.go:Getenv:164 → NO DATA
[TRACE] 10:49:59.179 terminal.go:Getenv(SHLVL) - 0s
[TRACE] 10:49:59.179 terminal.go:TemplateCache() - 1.1493ms
[TRACE] 10:49:59.179 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.179 terminal.go:StackCount() - 0s
[TRACE] 10:49:59.179 terminal.go:TemplateCache() - 1.1493ms
[DEBUG] 10:49:59.180 win32_windows.go:isWriteable:299 → not current user or in group
[DEBUG] 10:49:59.180 win32_windows.go:isWriteable:299 → not current user or in group
[DEBUG] 10:49:59.180 win32_windows.go:isWriteable:303 → current user is member of S-1-5-21-1774535685-1675276502-4012436528-1002
[DEBUG] 10:49:59.180 win32_windows.go:isWriteable:311 ↓
    WRITE_DAC
    WRITE_OWNER
    SYNCHRONIZE
    DELETE
    READ_CONTROL
[DEBUG] 10:49:59.180 win32_windows.go:isWriteable:313 → user has write access
[TRACE] 10:49:59.180 terminal_windows.go:DirIsWritable() - 1.8936ms
[TRACE] 10:49:59.180 terminal.go:TemplateCache() - 0s
[TRACE] 10:49:59.183 networks_windows.go:wifiNetwork() - 0s
[TRACE] 10:49:59.187 terminal.go:TemplateCache() - 0s
[TRACE] 10:49:59.187 terminal_windows_nix.go:BatteryState() - 10.3925ms
[DEBUG] 10:49:59.187 battery.go:Enabled:53 → NO DATA
[TRACE] 10:49:59.187 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.208 terminal.go:CommandPath:389 → C:\Users\Abel\AppData\Local\Programs\Git\cmd\git.exe
[TRACE] 10:49:59.208 terminal.go:CommandPath(git.exe) - 32.2977ms
[TRACE] 10:49:59.208 terminal.go:HasCommand(git.exe) - 32.2977ms
[DEBUG] 10:49:59.208 git.go:shouldDisplay:312 → fetch_bare_info: false
[TRACE] 10:49:59.208 terminal.go:Pwd() - 0s
[DEBUG] 10:49:59.209 terminal.go:ResolveSymlink:286 → C:\Users\Abel
[TRACE] 10:49:59.209 terminal.go:ResolveSymlink(C:\Users\Abel) - 895.1µs
[TRACE] 10:49:59.209 terminal.go:HasParentFilePath(.git) - 895.1µs
[DEBUG] 10:49:59.209 scm.go:shouldIgnoreRootRepository:150 → exclude_folders: []
[DEBUG] 10:49:59.209 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.209 git.go:Enabled:176 → fetch_user: false
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.209 git.go:Enabled:191 → cli
[DEBUG] 10:49:59.209 git.go:Enabled:196 → fetch_status: false
[DEBUG] 10:49:59.209 git.go:shouldIgnoreStatus:454 → ignore_status: []
[DEBUG] 10:49:59.209 terminal_windows.go:58 → C:\Users\Abel
[TRACE] 10:49:59.209 terminal.go:GOOS() - 0s
[DEBUG] 10:49:59.209 terminal.go:FileContent:301 ↓
    ref: refs/heads/main

[TRACE] 10:49:59.209 terminal.go:FileContent(C:\Users\Abel\.git/HEAD) - 675.6µs
[DEBUG] 10:49:59.209 git.go:setPrettyHEADName:805 → NO DATA
[DEBUG] 10:49:59.209 scm.go:formatBranch:114 → mapped_branches: map[]
[DEBUG] 10:49:59.209 scm.go:formatBranch:130 → full_branch_path: true
[DEBUG] 10:49:59.209 scm.go:formatBranch:136 → branch_max_length: 0
[DEBUG] 10:49:59.209 git.go:Enabled:207 → fetch_upstream_icon: true
[DEBUG] 10:49:59.209 git.go:getUpstreamIcon:537 → upstream_icons: map[]
[DEBUG] 10:49:59.209 git.go:getUpstreamIcon:558 → 
[TRACE] 10:49:59.209 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.209 text.go:Render:80 → rendering template:  ➜ ({{ .UpstreamIcon }}{{ .HEAD }}{{ if gt .StashCount 0 }}  {{ .StashCount }}{{ end }})
[TRACE] 10:49:59.209 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.210 text.go:Render:80 → rendering template:  {{ if not .Error }}{{ .Icon }}{{ .Percentage }}{{ end }}
[TRACE] 10:49:59.210 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.210 text.go:Render:80 → rendering template: <#000000>󰖳 {{.UserName}} {{.Path}}</>
[TRACE] 10:49:59.210 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.210 text.go:Render:80 → rendering template: {{ if eq .Type "wifi"}} {{.Name}}  {{ else if eq .Type "ethernet"}} {{ end }}
[TRACE] 10:49:59.210 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.210 text.go:Render:80 → rendering template:  {{ .CurrentDate | date .Format }}
[TRACE] 10:49:59.210 terminal.go:Flags() - 0s
[TRACE] 10:49:59.211 terminal.go:TemplateCache() - 0s
[TRACE] 10:49:59.213 terminal.go:TemplateCache() - 0s
[TRACE] 10:49:59.213 terminal.go:TemplateCache() - 0s
[TRACE] 10:49:59.213 terminal.go:TemplateCache() - 0s
[TRACE] 10:49:59.213 terminal.go:TemplateCache() - 0s
[ERROR] 10:49:59.214 terminal.go:FileContent:297 → open C:\Users\Abel\.git/logs/refs/stash: The system cannot find the file specified.
[TRACE] 10:49:59.214 terminal.go:FileContent(C:\Users\Abel\.git/logs/refs/stash) - 1.1631ms
[DEBUG] 10:49:59.214 text.go:Render:80 → rendering template: diamond
[TRACE] 10:49:59.214 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.214 text.go:Render:80 → rendering template: {{if eq "Charging" .State.String}}#40FF53{{end}}
[TRACE] 10:49:59.214 terminal.go:Flags() - 0s
[TRACE] 10:49:59.214 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.214 text.go:Render:80 → rendering template: {{if eq "Discharging" .State.String}}#000000{{end}}
[TRACE] 10:49:59.214 terminal.go:Flags() - 0s
[TRACE] 10:49:59.215 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.215 text.go:Render:80 → rendering template: {{if eq "Full" .State.String}}#ffffff{{end}}
[TRACE] 10:49:59.215 terminal.go:Flags() - 0s
[TRACE] 10:49:59.216 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.216 text.go:Render:80 → rendering template: {{if eq "Discharging" .State.String}}#ffffff{{end}}
[TRACE] 10:49:59.216 terminal.go:Flags() - 0s
[TRACE] 10:49:59.216 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.216 text.go:Render:80 → rendering template: powerline
[TRACE] 10:49:59.216 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.216 text.go:Render:80 → rendering template: {{if eq "Charging" .State.String}}#40FF53{{end}}
[TRACE] 10:49:59.216 terminal.go:Flags() - 0s
[TRACE] 10:49:59.216 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.216 text.go:Render:80 → rendering template: {{if eq "Discharging" .State.String}}#000000{{end}}
[TRACE] 10:49:59.216 terminal.go:Flags() - 0s
[TRACE] 10:49:59.217 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.217 text.go:Render:80 → rendering template: {{if eq "Full" .State.String}}#ffffff{{end}}
[TRACE] 10:49:59.217 terminal.go:Flags() - 0s
[TRACE] 10:49:59.217 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.217 text.go:Render:80 → rendering template: {{if eq "Discharging" .State.String}}#ffffff{{end}}
[TRACE] 10:49:59.217 terminal.go:Flags() - 0s
[TRACE] 10:49:59.217 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.217 text.go:Render:80 → rendering template: powerline
[TRACE] 10:49:59.217 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.217 text.go:Render:80 → rendering template: {{if eq "Charging" .State.String}}#40FF53{{end}}
[TRACE] 10:49:59.217 terminal.go:Flags() - 0s
[TRACE] 10:49:59.218 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.218 text.go:Render:80 → rendering template: {{if eq "Discharging" .State.String}}#000000{{end}}
[TRACE] 10:49:59.218 terminal.go:Flags() - 0s
[TRACE] 10:49:59.220 terminal.go:TemplateCache() - 0s
[DEBUG] 10:49:59.221 text.go:Render:80 → rendering template: diamond
[TRACE] 10:49:59.221 terminal.go:Flags() - 0s
[DEBUG] 10:49:59.221 text.go:Render:80 → rendering template: powerline
[TRACE] 10:49:59.221 terminal.go:Flags() - 0s
[TRACE] 10:49:59.221 terminal.go:Shell() - 0s
[TRACE] 10:49:59.221 terminal.go:Shell() - 0s
[TRACE] 10:49:59.221 terminal.go:Shell() - 0s
[DEBUG] 10:49:59.222 terminal.go:Getenv:164 → C:\Users\Abel\AppData\Local
[TRACE] 10:49:59.222 terminal.go:Getenv(LOCALAPPDATA) - 0s
[TRACE] 10:49:59.222 terminal.go:CachePath() - 0s
[TRACE] 10:49:59.222 terminal.go:Flags() - 0s
```

--- a/src/runtime/networks_windows.go
+++ b/src/runtime/networks_windows.go
@@ -144,7 +144,7 @@ type DOT11_SSID struct {
 	ucSSID      [DOT11_SSID_MAX_LENGTH]uint8
 }
 
-func (env *Terminal) getConnections() []*Connection {
+func (term *Terminal) getConnections() []*Connection {
 	var pIFTable2 *MIN_IF_TABLE2
 	_, _, _ = hGetIfTable2.Call(uintptr(unsafe.Pointer(&pIFTable2)))
 
@@ -153,7 +153,6 @@ func (env *Terminal) getConnections() []*Connection {
 	for i := 0; i < int(pIFTable2.NumEntries); i++ {
 		networkInterface := pIFTable2.Table[i]
 		alias := strings.TrimRight(syscall.UTF16ToString(networkInterface.Alias[:]), "\x00")
-		description := strings.TrimRight(syscall.UTF16ToString(networkInterface.Description[:]), "\x00")
 
 		if networkInterface.OperStatus != 1 || // not connected or functional
 			!networkInterface.InterfaceAndOperStatusFlags.HardwareInterface || // rule out software interfaces
@@ -181,9 +180,11 @@ func (env *Terminal) getConnections() []*Connection {
 			continue
 		}
 
+		term.DebugF("Found network interface: %s", alias)
+
 		network := &Connection{
 			Type:         connectionType,
-			Name:         description, // we want a relatable name, alias isn't that
+			Name:         alias,
 			TransmitRate: networkInterface.TransmitLinkSpeed,
 			ReceiveRate:  networkInterface.ReceiveLinkSpeed,
 			SSID:         ssid,
@@ -192,15 +193,15 @@ func (env *Terminal) getConnections() []*Connection {
 		networks = append(networks, network)
 	}
 
-	if wifi, err := env.wifiNetwork(); err == nil {
+	if wifi, err := term.wifiNetwork(); err == nil {
 		networks = append(networks, wifi)
 	}
 
 	return networks
 }
 
-func (env *Terminal) wifiNetwork() (*Connection, error) {
-	env.Trace(time.Now())
+func (term *Terminal) wifiNetwork() (*Connection, error) {
+	term.Trace(time.Now())
 	// Open handle
 	var pdwNegotiatedVersion uint32
 	var phClientHandle uint32
@@ -209,7 +210,6 @@ func (env *Terminal) wifiNetwork() (*Connection, error) {
 		return nil, err
 	}
 
-	// defer closing handle
 	defer func() {
 		_, _, _ = hWlanCloseHandle.Call(uintptr(phClientHandle), uintptr(unsafe.Pointer(nil)))
 	}()
@@ -229,12 +229,14 @@ func (env *Terminal) wifiNetwork() (*Connection, error) {
 		if network.isState != 1 {
 			continue
 		}
-		return env.parseNetworkInterface(network, phClientHandle)
+
+		return term.parseNetworkInterface(network, phClientHandle)
 	}
+
 	return nil, errors.New("Not connected")
 }
 
-func (env *Terminal) parseNetworkInterface(network *WLAN_INTERFACE_INFO, clientHandle uint32) (*Connection, error) {
+func (term *Terminal) parseNetworkInterface(network *WLAN_INTERFACE_INFO, clientHandle uint32) (*Connection, error) {
 	info := Connection{
 		Type: WIFI,
 	}
@@ -250,7 +252,7 @@ func (env *Terminal) parseNetworkInterface(network *WLAN_INTERFACE_INFO, clientH
 		uintptr(unsafe.Pointer(&wlanAttr)),
 		uintptr(unsafe.Pointer(nil)))
 	if e != 0 {
-		env.Error(err)
+		term.Error(err)
 		return &info, err
 	}
 
@@ -258,6 +260,8 @@ func (env *Terminal) parseNetworkInterface(network *WLAN_INTERFACE_INFO, clientH
 	ssid := wlanAttr.wlanAssociationAttributes.dot11Ssid
 	if ssid.uSSIDLength > 0 {
 		info.SSID = string(ssid.ucSSID[0:ssid.uSSIDLength])
+		info.Name = info.SSID
+		term.DebugF("Found wifi interface: %s", info.SSID)
 	}
 
 	info.TransmitRate = uint64(wlanAttr.wlanAssociationAttributes.ulTxRate / 1024)

--- a/src/runtime/terminal_unix.go
+++ b/src/runtime/terminal_unix.go
@@ -154,6 +154,7 @@ func (term *Terminal) Connection(_ ConnectionType) (*Connection, error) {
 	if len(term.networks) == 0 {
 		return nil, &NotImplemented{}
 	}
+
 	return nil, &NotImplemented{}
 }
 

--- a/src/runtime/terminal_windows.go
+++ b/src/runtime/terminal_windows.go
@@ -241,11 +241,13 @@ func (term *Terminal) Connection(connectionType ConnectionType) (*Connection, er
 		}
 		term.networks = networks
 	}
+
 	for _, network := range term.networks {
 		if network.Type == connectionType {
 			return network, nil
 		}
 	}
+
 	term.Error(fmt.Errorf("Network type '%s' not found", connectionType))
 	return nil, &NotImplemented{}
 }

--- a/website/docs/segments/system/connection.mdx
+++ b/website/docs/segments/system/connection.mdx
@@ -46,6 +46,5 @@ import Config from '@site/src/components/Config.js';
 | ------- | -------- | ------------------------------------------------------- |
 | `.Type` | `string` | the connection type type. Single values of `type` above |
 | `.Name` | `string` | the name of the connection                              |
-| `.SSID` | `string` | the SSID of the current wifi network                    |
 
 [templates]: /docs/configuration/templates

